File: backend/src/main/java/ch/clicktotranslate/translation/application/web/SegmentBundle.java
package ch.clicktotranslate.translation.application.web;

import org.jmolecules.ddd.annotation.ValueObject;

import ch.clicktotranslate.translation.domain.Language;

import java.time.Instant;

@ValueObject
public record SegmentBundle(String userId, String word, String sentence, Language sourceLanguage,
		Language targetLanguage, Source source, SourceMetadata sourceMetadata, Instant occurredAt) {
	public record Source(String type, String id, String title) {
	}

	public interface SourceMetadata {

	}

	public record GenericSourceMetadata(String url, String domain, Integer selectionOffset,
			Integer paragraphIndex) implements SourceMetadata {
	}

	public record YoutubeSourceMetadata(String url, String domain, String videoId,
			Integer timestampSeconds) implements SourceMetadata {
	}
}

File: browser-extension/domain/src/model/SegmentBundle.ts
import type { LanguageDto } from "./Language";
import type { SourceDto } from "./Source";
import type { SourceMetadataDto } from "./SourceMetadata";

// ISO-8601 UTC timestamp string (for example: 2026-02-22T12:00:00Z)
export type InstantString = string;

export interface SegmentBundleDto {
  word: string;
  sentence: string;
  sourceLanguage: LanguageDto;
  targetLanguage: LanguageDto;
  source: SourceDto;
  sourceMetadata?: SourceMetadataDto;
  occurredAt: InstantString;
}
